File: app/src/main/java/me/lx/sample/group/adapter/GroupedListAdapter.java
package me.lx.sample.group.adapter;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

import me.lx.rv.group.BaseViewHolder;
import me.lx.rv.group.GroupedRecyclerViewAdapter;
import me.lx.sample.R;
import me.lx.sample.group.entity.ChildEntity;
import me.lx.sample.group.entity.GroupEntity;

/**
 * 这是普通的分组Adapter 每一个组都有头部、尾部和子项。
 */
public class GroupedListAdapter extends GroupedRecyclerViewAdapter<GroupEntity,ChildEntity> {

//    public GroupedListAdapter(@NotNull List<GroupEntity> dataList) {
//        super(dataList);
//    }

    @NotNull
    @Override
    public List<ChildEntity> getChildrenList(GroupEntity entity) {
        return entity.getChildren();
    }

    @Override
    public int getChildrenCount(int groupPosition) {
        ArrayList<ChildEntity> children = getItems().get(groupPosition).getChildren();
        return children.size();
    }

    @Override
    public boolean hasHeader(int groupPosition) {
        return getItems().get(groupPosition).getChildren().size() > 0;
    }

    @Override
    public boolean hasFooter(int groupPosition) {
        return getItems().get(groupPosition).getChildren().size()>0;
    }

    @Override
    public int getHeaderLayout(int viewType) {
        return R.layout.adapter_header;
    }

    @Override
    public int getFooterLayout(int viewType) {
        return R.layout.adapter_footer;
    }

    @Override
    public int getChildLayout(int viewType) {
        return R.layout.adapter_child;
    }

    @Override
    public void onBindHeaderViewHolder(BaseViewHolder holder, int groupPosition) {
        GroupEntity entity = getItems().get(groupPosition);
       // System.out.println("onBindHeaderViewHolder()....22222...groupPosition="+groupPosition+"  text="+entity.getHeader());
        holder.setText(R.id.tv_header, entity.getHeader());
    }

    @Override
    public void onBindFooterViewHolder(BaseViewHolder holder, int groupPosition) {
        GroupEntity entity = getItems().get(groupPosition);
        holder.setText(R.id.tv_footer, entity.getFooter());
    }

    @Override
    public void onBindChildViewHolder(BaseViewHolder holder, int groupPosition, int childPosition) {
        ChildEntity entity = getItems().get(groupPosition).getChildren().get(childPosition);
        holder.setText(R.id.tv_child, entity.getChild());
    }


}

File: app/src/main/java/me/lx/sample/group/adapter/GroupedListAdapter.kt
package me.lx.sample.group.adapter

import androidx.databinding.ViewDataBinding
import me.lx.rv.group.GroupedRecyclerViewAdapter
import me.lx.sample.R
import me.lx.sample.group.entity.ChildEntity
import me.lx.sample.group.entity.GroupEntity

/**
 * 这是普通的分组Adapter 每一个组都有头部、尾部和子项。
 */
open class GroupedListAdapter : GroupedRecyclerViewAdapter<GroupEntity, ChildEntity>() {

    //    public GroupedListAdapter(@NotNull List<GroupEntity> dataList) {
    //        super(dataList);
    //    }

    override fun getChildrenList(groupItem: GroupEntity): List<ChildEntity> {
        return groupItem.childList
    }

    override fun getChildrenCount(groupPosition: Int, groupItem: GroupEntity): Int {
        return groupItem.childList.size
    }
    override fun hasHeader(groupPosition: Int): Boolean {
        return getItems()[groupPosition].childList.size > 0
    }

    override fun hasFooter(groupPosition: Int): Boolean {
        return getItems()[groupPosition].childList.size > 0
    }

    override fun getHeaderLayout(viewType: Int): Int {
        return R.layout.adapter_header
    }

    override fun getFooterLayout(viewType: Int): Int {
        return R.layout.adapter_footer
    }

    override fun getChildLayout(viewType: Int): Int {
        return R.layout.adapter_child
    }

    override fun onBindHeaderViewHolder(binding: ViewDataBinding, groupItem: GroupEntity, groupPosition: Int) {
    }

    override fun onBindFooterViewHolder(binding:ViewDataBinding,  groupItem: GroupEntity,groupPosition: Int) {
    }

    override fun onBindChildViewHolder(binding:ViewDataBinding, groupItem: GroupEntity, child: ChildEntity,groupPosition: Int,childPosition: Int) {
        // 而外设置变量
    }




}
